src/hazelcore/UHExp.re

```diff
@@ -308,10 +308,130 @@ let text_operand =
   | InvalidTextShape(t) => new_InvalidText(u_gen, t)
   };
 
-let associate = (seq: seq) => {
-  let skel_str = Skel.mk_skel_str(seq, Operators_Exp.to_parse_string);
-  let lexbuf = Lexing.from_string(skel_str);
-  SkelExprParser.skel_expr(SkelExprLexer.read, lexbuf);
+let associate = (seq: seq): skel => {
+  /**
+   * Shunting-yard opp algorithm according to the specification here
+   * https://en.wikipedia.org/wiki/Shunting-yard_algorithm#The_algorithm_in_detail
+   *
+   */
+  let rec go_seq =
+          (
+            skels: list(skel), /* List of skels to be combined into single output skel. */
```

Comment:
Let's name this something like `skel_stack`, just to make clear it's a stack

Reply:
Sure thing